### PR TITLE
Refactor: remove use of `escape-string-regexp`

### PR DESCRIPTION
**Pull request by fisker**
`getMaxContinuousCount` has tests covered #7262 .

Other changes are internal.

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/util.js [hunk 1]
```diff
@@ -682,37 +681,45 @@ function printNumber(rawNumber) {
...
-    new RegExp(`(${escapeStringRegexp(target)})+`, "g")
-  );
-
-  if (matches === null) {
-    return 0;
-  }
+  let max = 0;
+  const targetLength = target.length;
 
   const countPresent = new Map();
```
Comment (fisker) [inline src/common/util.js:708]: Original implementation use `Map`, not sure about the performance between `Set` and `Map`

**Comment by evilebottnawi** [comment]
Just a question - Why?

**Comment by fisker** [comment]
> Just a question - Why?

I think we used it in simple cases, don't really need it.


````js
	const delimiterRegex = Object.keys(DELIMITER_MAP).map(escape).join("|");
    new RegExp(	
      `^(${delimiterRegex})([^\\n]*)\\n(?:([\\s\\S]*?)\\n)?\\1[^\\n\\S]*(\\n|$)`	
    )
```

vs 

```js
const frontMatterRe = /^(-{3}|\+{3})([^\n]*)\n(?:([\S\s]*?)\n)?\1[^\S\n]*(\n|$)/;
```

Don't you think the second one easier to read?

**Comment by evilebottnawi** [comment]
@fisker both are hard to read :smile: to be honestly we need check perf

**Comment by fisker** [comment]
Yes, we need, but I don't know how 😢

**Comment by fisker** [comment]
Close this as it has been a long time , I assume other member don't agree on this, also there is a `Regexp.escape` proposal, I think we'll be able to use it when runtime supports it.

**Pull request closed without merging**